### PR TITLE
fix: actually give the larger incoming NGC custom packets via API to the client

### PR DESCRIPTION
**Pull request by zoff99**
This change is [Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2441)

### CODE REVIEW

File: toxcore/group_chats.c [hunk 1]
```diff
@@ -156,6 +156,11 @@ typedef enum Group_Sync_Flags {
     GF_STATE      = (1 << 2), // 4
 } Group_Sync_Flags;
 
+typedef enum Group_CustomPkts_Direction {
```
Comment (Green-Sky) [inline toxcore/group_chats.c:159]: I dont think we need an whole enum for 2 states.

Comment (JFreegman) [inline toxcore/group_chats.c:159]: Why `Pkts` and not just `Packets`?

File: toxcore/group_chats.c [hunk 2]
```diff
@@ -156,6 +156,11 @@ typedef enum Group_Sync_Flags {
...
+    GC_CUSTOMPKTS_SENDING      = (1 << 0), // 1
```
Comment (JFreegman) [inline toxcore/group_chats.c:160]: Why are we using bitmasks? There will only ever be two possibilities and never at the same time. I don't mind an enum but the values shouldn't be bit set.

File: toxcore/group_chats.c [hunk 3]
```diff
@@ -6057,6 +6068,7 @@ static bool handle_gc_lossless_packet(const GC_Session *c, GC_Chat *chat, const
     uint8_t packet_type;
     uint64_t message_id;
 
+
```
Comment (JFreegman) [inline toxcore/group_chats.c:6071]: Remove double empty lines

File: toxcore/group_chats.c [hunk 4]
```diff
@@ -4906,11 +4911,17 @@ static int handle_gc_private_message(const GC_Session *c, const GC_Chat *chat, c
 }
 
 /** @brief Returns false if a custom packet is too large. */
-static bool custom_gc_packet_length_is_valid(uint16_t length, bool lossless)
+static bool custom_gc_packet_length_is_valid(uint16_t length, bool lossless, Group_CustomPkts_Direction pkt_direction)
```
Comment (JFreegman) [inline toxcore/group_chats.c:4914]: This whole function should be rewritten with less nesting.

**Review by JFreegman — CHANGES_REQUESTED** [review]
Can you give a description for why this fix is necessary?

**Comment by zoff99** [comment]
i think this one is obsolete now in favor of #2535

**Pull request closed without merging**